src/pretix/base/models/orders.py

```diff
@@ -1219,6 +1220,38 @@ def create_transactions(self, is_new=False, positions=None, fees=None, dt_now=No
         _transactions_mark_order_clean(self.pk)
         return create
 
+    def tagged_secret(self, tag, secret_length=64):
+        return salted_hmac(key_salt=tag, value=self.secret, algorithm="sha256").hexdigest()[:secret_length]
+
+    @staticmethod
+    def get_with_secret_check(code, received_secret, tag, secret_length=64, qs=None):
+        if not qs:
+            qs = Order.objects
+        dummy = "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef"[:secret_length]
+        try:
+            order = qs.get(code=code)
+            if not hmac.compare_digest(
+                order.tagged_secret(tag, secret_length) if tag else order.secret,
+                received_secret[:secret_length].lower() if tag else received_secret.lower()
+            ) and not (
+                    # TODO: remove this clause after a while (compatibility with old secrets currently in flight)
```

Comment:
but not remove the `not tag` part, right? otherwise OrderDetailMixin would break

Reply:
i though about removing the whole part in parentheses, like this:
```diff
-              ) and not (
-                   # TODO: remove this clause after a while (compatibility with old secrets currently in flight)
-                    tag and hmac.compare_digest(
-                        hashlib.sha1(order.secret.lower().encode()).hexdigest(),
-                        received_secret.lower()
-                    )
```


why would this break OrderDetailMixin? from OrderDetailMixin, tag=None is passed in, so i assume the following condition would be checked, regardless of the second clause being removed or not (the `tag and` short-circuits the second clause to False anyway if no tag is present):

```
            if not hmac.compare_digest(
                 order.secret,
                 received_secret.lower()
            )
```

Reply:
oh, yes, my mistake